go/engine/device_add.go

```diff
@@ -100,7 +99,8 @@ func (e *DeviceAdd) Run(ctx *Context) (err error) {
 			e.G().Log.Debug("received secret phrase, adding to provisioner")
 			ks, err := libkb.NewKex2SecretFromPhrase(receivedSecret.Phrase)
 			if err != nil {
-				e.G().Log.Warning("DisplayAndPromptSecret error: %s", err)
+				e.G().Log.Warning("NewKex2SecretFromPhrase error: %s", err)
+				canceler()
```

Comment:
Why is the `defer func() { canceler() }` on line 110 still needed? And why wasn't it sufficient?